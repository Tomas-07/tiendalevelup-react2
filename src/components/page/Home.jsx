import React, { useState, useMemo } from 'react';
import Header from '../components/organisms/Header/Header';
import Footer from '../components/organisms/Footer/Footer';
import ProductCard from '../components/molecules/ProductCard/ProductCard';
import ProductFilter from '../components/molecules/ProductFilter/ProductFilter';
import { products, categories, priceRanges, formatPrice, filterProducts } from '../data/products';
import '../App.css';

const Home = () => {
  const [cartCount, setCartCount] = useState(0);
  const [searchTerm, setSearchTerm] = useState('');
  const [selectedCategory, setSelectedCategory] = useState('todas');
  const [selectedPrice, setSelectedPrice] = useState('todos');

  // Filtrar productos usando useMemo para optimización
  const filteredProducts = useMemo(() => {
    const filters = { searchTerm, category: selectedCategory, priceRange: selectedPrice };
    return filterProducts(products, filters);
  }, [searchTerm, selectedCategory, selectedPrice]);

  const handleAddToCart = (product) => {
    setCartCount(prev => prev + 1);
    console.log('Producto agregado al carrito:', product);
    // Aquí podrías agregar lógica para manejar el carrito
  };

  const handleNavClick = (section) => {
    const element = document.getElementById(section);
    if (element) {
      element.scrollIntoView({ behavior: 'smooth' });
    }
  };

  return (
    <div className="App">
      <Header 
        cartCount={cartCount} 
        onNavClick={handleNavClick}
      />
      
      <main>
        {/* SECCIÓN INICIO */}
        <section id="inicio" className="section">
          <h2>Bienvenido a Level-Up Gamer</h2>
          <p>Tu tienda online gamer en Chile 🚀</p>
          <p style={{ color: '#39FF14', marginTop: '1rem' }}>
            ¡Gana puntos LevelUp con cada compra y desbloquea descuentos exclusivos!
          </p>
        </section>

        {/* SECCIÓN CATÁLOGO */}
        <section id="catalogo" className="section">
          <h2>Nuestros Productos</h2>
          
          {/* Componente de Filtros */}
          <ProductFilter
            searchTerm={searchTerm}
            selectedCategory={selectedCategory}
            selectedPrice={selectedPrice}
            onSearchChange={setSearchTerm}
            onCategoryChange={setSelectedCategory}
            onPriceChange={setSelectedPrice}
            categories={categories}
            priceRanges={priceRanges}
          />

          {/* Contador de productos */}
          <div style={{ 
            textAlign: 'center', 
            marginBottom: '1rem',
            color: 'var(--light-gray)'
          }}>
            {filteredProducts.length} producto{filteredProducts.length !== 1 ? 's' : ''} encontrado{filteredProducts.length !== 1 ? 's' : ''}
          </div>

          {/* GRID DE PRODUCTOS */}
          <div className="productos">
            {filteredProducts.length > 0 ? (
              filteredProducts.map(product => (
                <ProductCard
                  key={product.id}
                  product={product}
                  onAddToCart={handleAddToCart}
                  formatPrice={formatPrice}
                />
              ))
            ) : (
              <div style={{ 
                gridColumn: '1 / -1', 
                textAlign: 'center', 
                padding: '3rem',
                color: 'var(--muted-gray)'
              }}>
                <h3>No se encontraron productos</h3>
                <p>Intenta con otros filtros de búsqueda</p>
              </div>
            )}
          </div>
        </section>

        {/* Las demás secciones se mantienen igual */}
        {/* SECCIÓN COMUNIDAD */}
        <section id="comunidad" className="section">
          <h2>Comunidad Gamer</h2>
          <p>Explora artículos, noticias y consejos para mejorar tu experiencia de juego.</p>
          
          <div className="comunidad-content">
            <h3>Últimas Noticias</h3>
            <div className="news-card">
              <h4>🎮 Próximos Lanzamientos 2025</h4>
              <p>Descubre los juegos más esperados del año y prepárate para las mejores experiencias gaming.</p>
            </div>
            
            <div className="news-card green-border">
              <h4>🏆 Consejos para Mejorar tu Setup</h4>
              <p>Optimiza tu estación de juego con nuestras recomendaciones de expertos.</p>
            </div>
          </div>
        </section>

        {/* SECCIÓN EVENTOS */}
        <section id="eventos" className="section">
          <h2>Eventos Gamer en Chile</h2>
          <p className="eventos-subtitle">¡Participa en eventos y gana puntos LevelUp!</p>
          
          <div className="eventos-container">
            <div className="eventos-card">
              <h3>📍 Próximos Eventos</h3>
              
              <div className="eventos-grid">
                <div className="evento-item">
                  <h4>🎮 Chile Gaming Expo 2025</h4>
                  <p><strong>Fecha:</strong> 15-17 Noviembre 2025</p>
                  <p><strong>Lugar:</strong> Centro de Eventos Movistar Arena, Santiago</p>
                  <p><strong>Puntos:</strong> 200 LevelUp por asistencia</p>
                </div>

                <div className="evento-item green-border">
                  <h4>🏆 Torneo League of Legends Regional</h4>
                  <p><strong>Fecha:</strong> 5 Diciembre 2025</p>
                  <p><strong>Lugar:</strong> Centro Cultural Gabriela Mistral, Santiago</p>
                  <p><strong>Puntos:</strong> 300 LevelUp por participación</p>
                </div>

                <div className="evento-item">
                  <h4>🎯 Meetup Gamers Valparaíso</h4>
                  <p><strong>Fecha:</strong> 20 Diciembre 2025</p>
                  <p><strong>Lugar:</strong> Puerto Valparaíso</p>
                  <p><strong>Puntos:</strong> 150 LevelUp por asistencia</p>
                </div>
              </div>
            </div>
          </div>
        </section>

        {/* SECCIÓN CONTACTO */}
        <section id="contacto" className="section">
          <h2>Contacto</h2>
          <p>¿Necesitas ayuda? Estamos aquí para ti.</p>
          
          <div className="contacto-container">
            <div className="contacto-card">
              <p>
                📧 Email: <a href="mailto:soporte@levelupgamer.cl" className="contact-link">
                  soporte@levelupgamer.cl
                </a>
              </p>
              <p>
                📱 WhatsApp: <a href="https://wa.me/56912345678" target="_blank" rel="noopener noreferrer" className="contact-link green">
                  +56 9 1234 5678
                </a>
              </p>
              <p>
                🕐 Horario de atención: Lunes a Viernes, 9:00 - 18:00 hrs
              </p>
              
              <a href="https://wa.me/56912345678" target="_blank" rel="noopener noreferrer" className="whatsapp-btn">
                💬 Chat con Soporte Técnico
              </a>
            </div>
          </div>
        </section>
      </main>

      <Footer />
    </div>
  );
};

export default Home;