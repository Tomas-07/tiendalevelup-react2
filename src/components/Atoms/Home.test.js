import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Home from '../page/Home';

// Mock de las funciones importadas
jest.mock('../Atoms/Validaciones', () => ({
  carrito: jest.fn((producto) => console.log('Añadido al carrito:', producto)),
  mostrarMensaje: jest.fn((mensaje) => console.log('Mensaje:', mensaje))
}));

const renderWithRouter = (component) => {
  return render(<BrowserRouter>{component}</BrowserRouter>);
};

describe('Home Component - 10 Pruebas Unitarias', () => {
  
  // 1. Prueba de renderizado inicial
  test('debe renderizar el componente Home correctamente', () => {
    const { container } = renderWithRouter(<Home />);
    expect(container).toBeInTheDocument();
  });

  // 2. Prueba del header
  test('debe renderizar el componente Header', () => {
    renderWithRouter(<Home/>);
    // Busca elementos típicos del header como enlaces o logo
    const links = screen.getAllByRole('link');
    expect(links.length).toBeGreaterThan(0);
  });

  // 3. Prueba del input de búsqueda
  test('debe tener un campo de búsqueda funcional', () => {
    renderWithRouter(<Home />);
    const searchInput = screen.getByPlaceholderText(/buscar/i);
    expect(searchInput).toBeInTheDocument();
  });

  // 4. Prueba de cambio en el input de búsqueda
  test('debe actualizar el estado de búsqueda al escribir', () => {
    renderWithRouter(<Home />);
    const searchInput = screen.getByPlaceholderText(/buscar/i);
    
    fireEvent.change(searchInput, { target: { value: 'laptop' } });
    expect(searchInput.value).toBe('laptop');
  });

  // 5. Prueba de carga de productos
  test('debe cargar productos desde localStorage al montar', async () => {
    // Mock de localStorage
    const mockProductos = JSON.stringify([
      { id: 1, nombre: 'Producto 1', precio: 100, imagen: 'img1.jpg' },
      { id: 2, nombre: 'Producto 2', precio: 200, imagen: 'img2.jpg' }
    ]);
    
    Storage.prototype.getItem = jest.fn(() => mockProductos);
    
    renderWithRouter(<Home />);
    
    await waitFor(() => {
      const productos = screen.queryAllByAltText(/producto/i);
      expect(productos.length).toBeGreaterThanOrEqual(0);
    });
  });

  
  test('debe filtrar productos según la búsqueda', async () => {
    const mockProductos = JSON.stringify([
      { id: 1, nombre: 'Laptop', precio: 1000, imagen: 'laptop.jpg' },
      { id: 2, nombre: 'Mouse', precio: 50, imagen: 'mouse.jpg' }
    ]);
    
    Storage.prototype.getItem = jest.fn(() => mockProductos);
    
    renderWithRouter(<Home />);
    const searchInput = screen.getByPlaceholderText(/buscar/i);
    
    fireEvent.change(searchInput, { target: { value: 'laptop' } });
    
    await waitFor(() => {
      expect(searchInput.value).toBe('laptop');
    });
  });


  test('debe manejar correctamente cuando no hay productos', () => {
    Storage.prototype.getItem = jest.fn(() => null);
    
    const { container } = renderWithRouter(<Home />);
    expect(container).toBeInTheDocument();
  });

  
  test('debe contener la estructura correcta con Header y productos', () => {
    const { container } = renderWithRouter(<Home />);
    
    
    expect(container.firstChild).toBeInTheDocument();
  });

  test('debe tener elementos HTML válidos', () => {
  Storage.prototype.getItem = jest.fn(() => JSON.stringify([
    { id: 1, nombre: 'Test', precio: 100, imagen: 'test.jpg' }
  ]));
  
  const { container } = renderWithRouter(<Home />);
  expect(container.querySelector('main')).toBeInTheDocument();
  expect(container.querySelector('section')).toBeInTheDocument();
});

  
  test('debe inicializar con un array vacío de productos y búsqueda vacía', () => {
    Storage.prototype.getItem = jest.fn(() => null);
    
    renderWithRouter(<Home />);
    const searchInput = screen.getByPlaceholderText(/buscar/i);
    
    expect(searchInput.value).toBe('');
  });
});
