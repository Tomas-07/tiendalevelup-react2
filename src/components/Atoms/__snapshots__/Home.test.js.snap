// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Home Component - 10 Pruebas Unitarias debe coincidir con el snapshot 1`] = `
<div>
  <main>
    <section
      id="inicio"
    >
      <h2>
        Bienvenido a Level-Up Gamer
      </h2>
      <p>
        Tu tienda online gamer en Chile 🚀
      </p>
      <p
        style="color: rgb(57, 255, 20); margin-top: 1rem;"
      >
        ¡Gana puntos LevelUp con cada compra y desbloquea descuentos exclusivos!
      </p>
    </section>
    <section
      id="catalogo"
    >
      <h2>
        Nuestros Productos
      </h2>
      <div
        class="filtros-container"
        style="max-width: 1200px; margin: 0px auto 2rem; display: grid; grid-template-columns: repeat(auto-fit, minmax(250px, 1fr)); gap: 1rem; padding: 0px 2rem;"
      >
        <div>
          <label
            for="busqueda"
            style="display: block; color: rgb(211, 211, 211); margin-bottom: 0.5rem; font-weight: 600;"
          >
            🔍 Buscar productos
          </label>
          <input
            id="busqueda"
            placeholder="Buscar por nombre..."
            style="width: 100%; padding: 0.8rem; border-radius: 8px; border: 1px solid rgb(34, 34, 34); background: rgb(26, 26, 26); color: rgb(255, 255, 255); font-size: 1rem;"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="filtro-categoria"
            style="display: block; color: rgb(211, 211, 211); margin-bottom: 0.5rem; font-weight: 600;"
          >
            📦 Categoría
          </label>
          <select
            id="filtro-categoria"
            style="width: 100%; padding: 0.8rem; border-radius: 8px; border: 1px solid rgb(34, 34, 34); background: rgb(26, 26, 26); color: rgb(255, 255, 255); font-size: 1rem; cursor: pointer;"
          >
            <option
              value="todas"
            >
              Todas las categorías
            </option>
            <option
              value="juegos-mesa"
            >
              Juegos de Mesa
            </option>
            <option
              value="accesorios"
            >
              Accesorios
            </option>
            <option
              value="consolas"
            >
              Consolas
            </option>
            <option
              value="computadores"
            >
              Computadores Gamers
            </option>
            <option
              value="sillas"
            >
              Sillas Gamers
            </option>
            <option
              value="mouse"
            >
              Mouse
            </option>
            <option
              value="mousepad"
            >
              Mousepad
            </option>
            <option
              value="poleras"
            >
              Poleras Personalizadas
            </option>
            <option
              value="polerones"
            >
              Polerones Gamers
            </option>
          </select>
        </div>
        <div>
          <label
            for="filtro-precio"
            style="display: block; color: rgb(211, 211, 211); margin-bottom: 0.5rem; font-weight: 600;"
          >
            💰 Rango de precio
          </label>
          <select
            id="filtro-precio"
            style="width: 100%; padding: 0.8rem; border-radius: 8px; border: 1px solid rgb(34, 34, 34); background: rgb(26, 26, 26); color: rgb(255, 255, 255); font-size: 1rem; cursor: pointer;"
          >
            <option
              value="todos"
            >
              Todos los precios
            </option>
            <option
              value="bajo"
            >
              Menos de $50.000
            </option>
            <option
              value="medio"
            >
              $50.000 - $200.000
            </option>
            <option
              value="alto"
            >
              Más de $200.000
            </option>
          </select>
        </div>
      </div>
      <div
        class="productos"
      >
        <article
          class="producto"
          data-categoria="juegos-mesa"
          data-codigo="JM001"
          data-precio="29990"
        >
          <img
            alt="Catan"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTVZG1G_ZdkpPVHE9O8kqSNWPRHjKC3Vz9Sag&s"
          />
          <h3>
            Catan
          </h3>
          <p
            class="precio"
          >
            $29.990 CLP
          </p>
          <p
            class="descripcion"
          >
            Un clásico juego de estrategia para 3-4 jugadores.
          </p>
          <button
            class="btn-agregar"
          >
            Agregar al carrito
          </button>
        </article>
        <article
          class="producto"
          data-categoria="juegos-mesa"
          data-codigo="JM002"
          data-precio="24990"
        >
          <img
            alt="Carcassonne"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRUNb5P5Cj9UCEVmLua1VI7Z81P79V4DtMu0Q&s"
          />
          <h3>
            Carcassonne
          </h3>
          <p
            class="precio"
          >
            $24.990 CLP
          </p>
          <p
            class="descripcion"
          >
            Juego de fichas y construcción de paisajes medievales.
          </p>
          <button
            class="btn-agregar"
          >
            Agregar al carrito
          </button>
        </article>
        <article
          class="producto"
          data-categoria="consolas"
          data-codigo="CO001"
          data-precio="549990"
        >
          <img
            alt="PlayStation 5"
            src="https://gsmpro.cl/cdn/shop/articles/16991bf65a7a52901d78b55fa36bddc8.jpg?v=1737343320"
          />
          <h3>
            PlayStation 5
          </h3>
          <p
            class="precio"
          >
            $549.990 CLP
          </p>
          <p
            class="descripcion"
          >
            Consola de nueva generación con gráficos impresionantes.
          </p>
          <a
            class="btn-detalles"
            data-discover="true"
            href="/play"
          >
            Ver detalles
          </a>
          <button
            class="btn-agregar"
          >
            Agregar al carrito
          </button>
        </article>
        <article
          class="producto"
          data-categoria="computadores"
          data-codigo="CG001"
          data-precio="1299990"
        >
          <img
            alt="PC Gamer ASUS ROG Strix"
            src="https://dlcdnwebimgs.asus.com/files/media/6C1CAB30-D5C6-4D6E-90DC-B6A088360E12/V1/img/frame/01.jpg"
          />
          <h3>
            PC Gamer ASUS ROG Strix
          </h3>
          <p
            class="precio"
          >
            $1.299.990 CLP
          </p>
          <p
            class="descripcion"
          >
            Potente equipo para los gamers más exigentes.
          </p>
          <button
            class="btn-agregar"
          >
            Agregar al carrito
          </button>
        </article>
        <article
          class="producto"
          data-categoria="sillas"
          data-codigo="SG001"
          data-precio="349990"
        >
          <img
            alt="Silla Gamer Secretlab Titan"
            src="https://m.media-amazon.com/images/I/51ajrSvAdiL.jpg"
          />
          <h3>
            Silla Gamer Secretlab Titan
          </h3>
          <p
            class="precio"
          >
            $349.990 CLP
          </p>
          <p
            class="descripcion"
          >
            Ergonómica, cómoda y ajustable para largas sesiones de juego.
          </p>
          <button
            class="btn-agregar"
          >
            Agregar al carrito
          </button>
        </article>
        <article
          class="producto"
          data-categoria="mouse"
          data-codigo="MS001"
          data-precio="49990"
        >
          <img
            alt="Mouse Gamer Logitech G502 HERO"
            src="https://m.media-amazon.com/images/I/61mpMH5TzkL._AC_UF894,1000_QL80_.jpg"
          />
          <h3>
            Mouse Gamer Logitech G502 HERO
          </h3>
          <p
            class="precio"
          >
            $49.990 CLP
          </p>
          <p
            class="descripcion"
          >
            Sensor de alta precisión y botones personalizables.
          </p>
          <button
            class="btn-agregar"
          >
            Agregar al carrito
          </button>
        </article>
        <article
          class="producto"
          data-categoria="mousepad"
          data-codigo="MP001"
          data-precio="29990"
        >
          <img
            alt="Mousepad Razer Goliathus Extended Chroma"
            src="https://assets2.razerzone.com/images/pnx.assets/f024c732d60734e43b0c95945683c3cc/razer-goliathus-extended-chroma-size.jpg"
          />
          <h3>
            Mousepad Razer Goliathus Extended Chroma
          </h3>
          <p
            class="precio"
          >
            $29.990 CLP
          </p>
          <p
            class="descripcion"
          >
            Área de juego amplia con iluminación RGB personalizable.
          </p>
          <button
            class="btn-agregar"
          >
            Agregar al carrito
          </button>
        </article>
      </div>
    </section>
    <section
      id="comunidad"
    >
      <h2>
        Comunidad Gamer
      </h2>
      <p>
        Explora artículos, noticias y consejos para mejorar tu experiencia de juego.
      </p>
      <div
        style="max-width: 800px; margin: 2rem auto; text-align: left; padding: 0px 2rem;"
      >
        <h3
          style="color: rgb(57, 255, 20); margin-bottom: 1rem;"
        >
          Últimas Noticias
        </h3>
        <div
          style="background: rgb(17, 17, 17); padding: 1.5rem; border-radius: 12px; margin-bottom: 1rem; border-left: 4px solid rgb(30, 144, 255);"
        >
          <h4
            style="color: rgb(30, 144, 255); margin-bottom: 0.5rem;"
          >
            🎮 Próximos Lanzamientos 2025
          </h4>
          <p
            style="color: rgb(211, 211, 211);"
          >
            Descubre los juegos más esperados del año y prepárate para las mejores experiencias gaming.
          </p>
        </div>
        <div
          style="background: rgb(17, 17, 17); padding: 1.5rem; border-radius: 12px; margin-bottom: 1rem; border-left: 4px solid rgb(57, 255, 20);"
        >
          <h4
            style="color: rgb(57, 255, 20); margin-bottom: 0.5rem;"
          >
            🏆 Consejos para Mejorar tu Setup
          </h4>
          <p
            style="color: rgb(211, 211, 211);"
          >
            Optimiza tu estación de juego con nuestras recomendaciones de expertos.
          </p>
        </div>
      </div>
    </section>
    <section
      id="eventos"
    >
      <h2>
        Eventos Gamer en Chile
      </h2>
      <p
        style="margin-bottom: 2rem;"
      >
        ¡Participa en eventos y gana puntos LevelUp!
      </p>
      <div
        style="max-width: 1000px; margin: 0px auto;"
      >
        <div
          style="background: rgb(17, 17, 17); padding: 2rem; border-radius: 16px; border: 1px solid rgb(34, 34, 34);"
        >
          <h3
            style="color: rgb(30, 144, 255); margin-bottom: 1.5rem;"
          >
            📍 Próximos Eventos
          </h3>
          <div
            style="display: grid; gap: 1rem;"
          >
            <div
              style="background: rgb(26, 26, 26); padding: 1.5rem; border-radius: 12px; border-left: 4px solid rgb(30, 144, 255);"
            >
              <h4
                style="color: rgb(57, 255, 20); margin-bottom: 0.5rem;"
              >
                🎮 Chile Gaming Expo 2025
              </h4>
              <p
                style="color: rgb(211, 211, 211); margin: 0.5rem 0px;"
              >
                <strong>
                  Fecha:
                </strong>
                 15-17 Noviembre 2025
              </p>
              <p
                style="color: rgb(211, 211, 211); margin: 0.5rem 0px;"
              >
                <strong>
                  Lugar:
                </strong>
                 Centro de Eventos Movistar Arena, Santiago
              </p>
              <p
                style="color: rgb(211, 211, 211); margin: 0.5rem 0px;"
              >
                <strong>
                  Puntos:
                </strong>
                 200 LevelUp por asistencia
              </p>
            </div>
            <div
              style="background: rgb(26, 26, 26); padding: 1.5rem; border-radius: 12px; border-left: 4px solid rgb(57, 255, 20);"
            >
              <h4
                style="color: rgb(57, 255, 20); margin-bottom: 0.5rem;"
              >
                🏆 Torneo League of Legends Regional
              </h4>
              <p
                style="color: rgb(211, 211, 211); margin: 0.5rem 0px;"
              >
                <strong>
                  Fecha:
                </strong>
                 5 Diciembre 2025
              </p>
              <p
                style="color: rgb(211, 211, 211); margin: 0.5rem 0px;"
              >
                <strong>
                  Lugar:
                </strong>
                 Centro Cultural Gabriela Mistral, Santiago
              </p>
              <p
                style="color: rgb(211, 211, 211); margin: 0.5rem 0px;"
              >
                <strong>
                  Puntos:
                </strong>
                 300 LevelUp por participación
              </p>
            </div>
            <div
              style="background: rgb(26, 26, 26); padding: 1.5rem; border-radius: 12px; border-left: 4px solid rgb(30, 144, 255);"
            >
              <h4
                style="color: rgb(57, 255, 20); margin-bottom: 0.5rem;"
              >
                🎯 Meetup Gamers Valparaíso
              </h4>
              <p
                style="color: rgb(211, 211, 211); margin: 0.5rem 0px;"
              >
                <strong>
                  Fecha:
                </strong>
                 20 Diciembre 2025
              </p>
              <p
                style="color: rgb(211, 211, 211); margin: 0.5rem 0px;"
              >
                <strong>
                  Lugar:
                </strong>
                 Puerto Valparaíso
              </p>
              <p
                style="color: rgb(211, 211, 211); margin: 0.5rem 0px;"
              >
                <strong>
                  Puntos:
                </strong>
                 150 LevelUp por asistencia
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      id="contacto"
    >
      <h2>
        Contacto
      </h2>
      <p>
        ¿Necesitas ayuda? Estamos aquí para ti.
      </p>
      <div
        style="max-width: 600px; margin: 2rem auto; text-align: left; padding: 0px 2rem;"
      >
        <div
          style="background: rgb(17, 17, 17); padding: 2rem; border-radius: 12px; border: 1px solid rgb(34, 34, 34);"
        >
          <p
            style="margin-bottom: 1rem;"
          >
            📧 Email: 
            <a
              href="mailto:soporte@levelupgamer.cl"
              style="color: rgb(30, 144, 255);"
            >
              soporte@levelupgamer.cl
            </a>
          </p>
          <p
            style="margin-bottom: 1rem;"
          >
            📱 WhatsApp: 
            <a
              href="https://wa.me/56912345678"
              rel="noopener noreferrer"
              style="color: rgb(57, 255, 20);"
              target="_blank"
            >
              +56 9 1234 5678
            </a>
          </p>
          <p
            style="margin-bottom: 1rem;"
          >
            🕐 Horario de atención: Lunes a Viernes, 9:00 - 18:00 hrs
          </p>
          <a
            href="https://wa.me/56912345678"
            rel="noopener noreferrer"
            style="display: inline-block; background: rgb(37, 211, 102); color: white; padding: 0.8rem 1.5rem; border-radius: 8px; text-decoration: none; font-weight: bold; margin-top: 1rem;"
            target="_blank"
          >
            💬 Chat con Soporte Técnico
          </a>
        </div>
      </div>
    </section>
  </main>
</div>
`;
